File: Bronze/1037/1037.java
// [1037] 약수
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];

        String[] s = br.readLine().split(" ");
        for (int i = 0; i < n; i++)
            arr[i] = Integer.parseInt(s[i]);

        Arrays.sort(arr);
        System.out.println(arr[0] * arr[n-1]);
    }
}

File: Bronze/1037/1037.cpp
// [1037] 약수
#include <iostream>
#include <algorithm>
using namespace std;

int main(void)
{
    int n, arr[51];
    cin >> n;
    
    for (int i = 0; i < n; i++)
        cin >> arr[i];
    
    sort(arr, arr+n);
    cout << arr[0] * arr[n-1] << "\n";
}
